src/_guides/libraries/library-tour.md

```diff
@@ -2333,61 +2316,49 @@ the object's `toJson()` method.
 
 Use `UTF8.decode()` to decode UTF8-encoded bytes to a Dart string:
 
-<!-- library-tour/decode_utf-8/bin/main.dart -->
+<?code-excerpt "test/library_tour/convert_test.dart (UTF8-decode)" replace="/ \/\/line-br.*//g"?>
 {% prettify dart %}
-import 'dart:convert' show UTF8;
+List<int> utf8Bytes = [
+  0xc3, 0x8e, 0xc3, 0xb1, 0xc5, 0xa3, 0xc3, 0xa9,
+  0x72, 0xc3, 0xb1, 0xc3, 0xa5, 0xc5, 0xa3, 0xc3,
+  0xae, 0xc3, 0xb6, 0xc3, 0xb1, 0xc3, 0xa5, 0xc4,
+  0xbc, 0xc3, 0xae, 0xc5, 0xbe, 0xc3, 0xa5, 0xc5,
+  0xa3, 0xc3, 0xae, 0xe1, 0xbb, 0x9d, 0xc3, 0xb1
+];
 
-main() {
-  var string = UTF8.decode([
-    0xc3, 0x8e, 0xc3, 0xb1, 0xc5, 0xa3, 0xc3, 0xa9,
-    0x72, 0xc3, 0xb1, 0xc3, 0xa5, 0xc5, 0xa3, 0xc3,
-    0xae, 0xc3, 0xb6, 0xc3, 0xb1, 0xc3, 0xa5, 0xc4,
-    0xbc, 0xc3, 0xae, 0xc5, 0xbe, 0xc3, 0xa5, 0xc5,
-    0xa3, 0xc3, 0xae, 0xe1, 0xbb, 0x9d, 0xc3, 0xb1
-  ]);
-  print(string); // 'Îñţérñåţîöñåļîžåţîờñ'
-}
+var funnyWord = UTF8.decode(utf8Bytes);
+
+assert(funnyWord == 'Îñţérñåţîöñåļîžåţîờñ');
 {% endprettify %}
 
 To convert a stream of UTF-8 characters into a Dart string, specify
 `UTF8.decoder` to the Stream `transform()` method:
 
-<!-- library-tour/read-file/read_file.dart -->
+<?code-excerpt "test/library_tour/io_test.dart (UTF8-decoder)" replace="/UTF8.decoder/[!$&!]/g"?>
 {% prettify dart %}
 var lines = inputStream
-    .transform(UTF8.decoder)
+    .transform([!UTF8.decoder!])
     .transform(new LineSplitter());
 try {
   await for (var line in lines) {
     print('Got ${line.length} characters from stream');
   }
+  print('file is now closed');
+} catch (e) {
+  print(e);
+}
 {% endprettify %}
 
 Use `UTF8.encode()` to encode a Dart string as a list of UTF8-encoded
 bytes:
 
-<!-- library-tour/encode_utf-8/bin/main.dart -->
+<?code-excerpt "test/library_tour/convert_test.dart (UTF8-encode)" replace="/ \/\/line-br.*//g"?>
 {% prettify dart %}
-import 'dart:convert' show UTF8;
-
-main() {
-  List<int> expected = [
-    0xc3, 0x8e, 0xc3, 0xb1, 0xc5, 0xa3, 0xc3, 0xa9,
-    0x72, 0xc3, 0xb1, 0xc3, 0xa5, 0xc5, 0xa3, 0xc3,
-    0xae, 0xc3, 0xb6, 0xc3, 0xb1, 0xc3, 0xa5, 0xc4,
-    0xbc, 0xc3, 0xae, 0xc5, 0xbe, 0xc3, 0xa5, 0xc5,
-    0xa3, 0xc3, 0xae, 0xe1, 0xbb, 0x9d, 0xc3, 0xb1
-  ];
```

Comment:
No need to repeat this long array. In the page it is displayed just a few lines above in the previous excerpt.